### PR TITLE
Backend Implementation of setting course module dates on basis of course start date/open-date

### PR DESCRIPTION
**Pull request by saad-waqar**
Added new app openedx/features/cms 
changed re_run creation task. Added method that performs operations post-creation 
set module release and due dates

### CODE REVIEW

File: cms/djangoapps/contentstore/tasks.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def rerun_course(source_course_key_string, destination_course_key_string, user_i
         copy_course_videos(source_course_key, destination_course_key)
 
         # initialize course settings like tags stc
-        initialize_course_settings(source_course_key, destination_course_key)
+        apply_post_rerun_creation_tasks(source_course_key, destination_course_key, User.objects.get(id=user_id))
```
Comment (emraan004) [inline cms/djangoapps/contentstore/tasks.py]: is this a good idea passing query(`User.objects.get(id=user_id)`) as parameter?

File: cms/djangoapps/contentstore/tasks.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def rerun_course(source_course_key_string, destination_course_key_string, user_i
         copy_course_videos(source_course_key, destination_course_key)
 
         # initialize course settings like tags stc
```
Comment (emraan004) [inline cms/djangoapps/contentstore/tasks.py]: let's update this comment according to our change

File: cms/djangoapps/contentstore/tasks.py [hunk 3]
```diff
@@ -52,8 +52,8 @@ def rerun_course(source_course_key_string, destination_course_key_string, user_i
         # call edxval to attach videos to the rerun
         copy_course_videos(source_course_key, destination_course_key)
 
-        # initialize course settings like tags stc
-        initialize_course_settings(source_course_key, destination_course_key)
+        # Apply post re run creation tasks
```
Comment (ahmed-faraz46) [inline cms/djangoapps/contentstore/tasks.py]: Let's, please remove this comment or make is more descriptive as it's just function's name.

File: openedx/features/cms/helpers.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    if re_run.start == CourseFields.start.default:
+        return
+
+    source_course = get_course_by_id(source_course_key)
+
+    # Set course re-run module start and due dates according to the source course
+    set_rerun_course_module_dates(source_course, re_run, user)
+
+
+def set_rerun_course_module_dates(source_course, re_run, user):
```
Comment (ahmed-faraz46) [inline openedx/features/cms/helpers.py]: Function documentation, please.

**Pull request merged**